lib/custom/src/MShop/Service/Provider/Payment/Stripe.php

```diff
@@ -33,20 +129,276 @@ protected function getConfigPrefix()
 
 
 	/**
-	 * Returns the value for the given configuration key
+	 * Returns the Omnipay gateway provider object.
+	 *
+	 * @return \Omnipay\Common\GatewayInterface Gateway provider object
+	 */
+	protected function getProvider()
+	{
+		$config = $this->getServiceItem()->getConfig();
+		$config['apiKey'] = $this->getValue('apiKey','');
```

Comment:
This is unnecessary as 'apiKey' is already set in $config

Reply:
I did not clearly understanding. Why it is not necessary? If we do not pass it there, we will get "Invalid API Key provided:" from the Stripe. We have that key in the $config, but as $config['stripe.apiKey'];
Do you mean we have to write
$config['apiKey'] = $config['stripe.apiKey'];
instead of 
$config['apiKey'] = $this->getValue('apiKey','');
?

Reply:
You did not provide an API key. You need to provide your API key in the Authorization header, using Bearer auth (e.g. 'Authorization: Bearer YOUR_SECRET_KEY'). See https://stripe.com/docs/api#authentication for details, or we can help at https://support.stripe.com/.

Reply:
OK, got it. Then it's fine.